components/sidebar/sidebar_channel_list/sidebar_channel_list.tsx

```diff
@@ -396,8 +409,24 @@ export default class SidebarChannelList extends React.PureComponent<Props, State
 
         if (result.reason === 'DROP' && result.destination) {
             if (result.type === 'SIDEBAR_CHANNEL') {
-                this.props.actions.moveChannelInSidebar(result.destination.droppableId, result.draggableId, result.destination.index);
-                trackEvent('ui', 'ui_sidebar_dragdrop_dropped_channel');
+                let channelsToMove = [result.draggableId];
+                if (this.props.selectedChannelIds.length) {
+                    // Filter out channels that can't go in the category specified
+                    const targetCategory = this.props.categories.find((category) => category.id === result.destination?.droppableId);
```

Comment:
Perhaps this logic could be handled by `moveChannelsInSidebar`? I wonder if we might have easier access to some of this data in the `action`. Also, it might be easier to test if we do that since we can just give `moveChannelsInSidebar` an arbitrary list of IDs and see what happens.

Reply:
Yeah I think that's a smart refactor, there's a lot of logic in the component here. I'll make the change.